jetty-jmx/src/main/java/org/eclipse/jetty/jmx/MBeanContainer.java

```diff
@@ -293,4 +494,87 @@ private void unregister(ObjectName objectName)
             LOG.warn(x);
         }
     }
+
+    private class MetaData
+    {
+        final ManagedObject _managedObject;
+        final List<ManagedAttribute> _attributes = new ArrayList<>();
+        final List<Method> _getters = new ArrayList<>();
+        final List<ManagedOperation> _operations = new ArrayList<>();
+        final List<Method> _methods = new ArrayList<>();
+
+        MetaData(Class<?> oClass)
+        {
+            // Process Type Annotations
+            _managedObject = oClass.getAnnotation(ManagedObject.class);
+
+            // Process Method Annotations on this class
+            for (Method method : oClass.getDeclaredMethods())
```

Comment:
I think we should only loop over the public methods, not the declared ones. Again, we don't call `setAccessible()` so if we get a private method we won't be able to call it anyway.